### ISSUE TITLE
RegExp messes with "disallowSpaceBeforeBinaryOperators"

### ISSUE THREAD

**Issue by donaldpipowitch** [jscs-dev/node-jscs#849]
This throws an error:
````
Operator , should stick to previous expression at...:
    71 |      let url = input.replace(/^.$/, '$1');
---------------------------------------^
````

I'm pretty sure this is a bug as this works without errors:
````
let url = input.replace('foo', '$1');
````

`.jscsrc`:
````
{
  "disallowSpaceBeforeBinaryOperators": [
    ","
  ]
}
````

**Comment by ivanvotti** [comment]
Having the same problem (`"esnext": true`):
```
Operator , should stick to previous expression at app/controllers/order.js :
    55 |    return this.get('status').replace(/_/g, ' ');
-----------------------------------------------^
```

**Comment by alawatthe** [comment]
I think this looks like a duplicate of https://github.com/jscs-dev/node-jscs/issues/685.

**Comment by zxqfox** [comment]
Yeah, seems like. One more was closed already: #777
